include/particle.tcc

```diff
@@ -517,6 +540,45 @@ bool mpm::Particle<Tdim, Tnphases>::compute_stress(unsigned phase) {
   return status;
 }
 
+// Compute pore pressure
+template <unsigned Tdim, unsigned Tnphases>
+bool mpm::Particle<Tdim, Tnphases>::compute_pore_pressure(
+    unsigned solid_skeleton, unsigned pore_fluid, double dt) {
+  bool status = true;
+  try {
+    // Check if material ptr is valid
+    if (material_.at(solid_skeleton) != nullptr &&
+        material_.at(pore_fluid) != nullptr) {
+      // Bulk modulus of fluid
+      double K = material_.at(pore_fluid)->property("bulk_modulus");
+      // Compute strain rate of the particle
+      // Solid skeleton
+      Eigen::VectorXd strain_rate_solid =
+          cell_->compute_strain_rate(bmatrix_, solid_skeleton);
+      // Pore fluid
+      Eigen::VectorXd strain_rate_fluid =
+          cell_->compute_strain_rate(bmatrix_, pore_fluid);
+      // Calculate pore pressure
+      double dpressure = K / porosity_ *
```

Comment:
remove solid volume fraction.

Reply:
missing (-) sign

Reply:
> remove solid volume fraction.

Now the velocities stored are the real velocities of solid and fluid, so I think we need the volume fraction.